jobs/kpi-forecasting/kpi-forecasting/metric_hub.py

```diff
@@ -0,0 +1,101 @@
+import pandas as pd
+
+from dataclasses import dataclass
+from google.cloud import bigquery
+from mozanalysis.config import ConfigLoader
+from textwrap import dedent
+from typing import Optional
+
+
+@dataclass
+class MetricHub:
+    """
+    Programatically get Metric Hub metrics from Big Query.
+    See https://mozilla.github.io/metric-hub/metrics/ for a list of metrics.
+
+    Args:
+        app_name (str): The Metric Hub app name for the metric.
+        slug (str): The Metric Hub slug for the metric.
+        start_date (str): A 'YYYY-MM-DD' formatted-string that specifies the first
+            date the metric should be queried.
+        end_date (str): A 'YYYY-MM-DD' formatted-string that specifies the last
+            date the metric should be queried.
+        alias (str): An alias for the metric. For example, 'DAU' instead of
+            'daily_active_users'.
+        project (str): The Big Query project to use when establishing a connection
+            to the Big Query client.
+    """
+
+    app_name: str
+    slug: str
+    start_date: str
```

Comment:
some validation on the date fields would be nice since this is user input and we expect a specific format.

Reply:
Good call. I'm now coercing these args to dates using `pd.to_datetime`. Pandas will undoubtedly catch more edge cases than I could write out.